File: app/Models/Culture.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Culture extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'nom_commun',
        'nom_scientifique',
        'type',
        'saison',
        'ph_sol_min',
        'ph_sol_max',
        'temp_min',
        'temp_max',
        'besoin_eau_cycle',
        'soil_type',
        'conseils',
        'region'
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'type' => \App\Enums\CultureTypeEnum::class,
            'saison' => \App\Enums\SaisonEnum::class,
            'soil_type' => \App\Enums\SoilTypeEnum::class,
            'ph_sol_min' => 'float',
            'ph_sol_max' => 'float',
            'temp_min' => 'integer',
            'temp_max' => 'integer',
            'besoin_eau_cycle' => 'integer',
        ];
    }

    /**
     * Get the products for this culture.
     */
    public function products()
    {
        return $this->belongsToMany(Product::class, 'culture_product')
            ->withPivot('dosage_specifique', 'notes')
            ->withTimestamps();
    }

    /**
     * Get the parcels for this culture.
     */
    public function parcels()
    {
        return $this->hasMany(Parcel::class);
    }
}

File: mobile/lib/src/models/culture.dart
import 'package:equatable/equatable.dart';

import 'product.dart';

/// Mirrors App\Http\Resources\CultureResource.
class Culture extends Equatable {
  const Culture({
    required this.id,
    required this.nomCommun,
    this.nomScientifique,
    this.type,
    this.saison,
    this.region,
    this.soilType,
    this.phSolMin,
    this.phSolMax,
    this.tempMin,
    this.tempMax,
    this.besoinEauCycle,
    this.conseils,
    this.products = const [],
  });

  final int id;
  final String nomCommun;
  final String? nomScientifique;
  final String? type;
  final String? saison;
  final String? region;
  final String? soilType;
  final double? phSolMin;
  final double? phSolMax;
  final double? tempMin;
  final double? tempMax;
  final double? besoinEauCycle;
  final String? conseils;
  final List<Product> products;

  factory Culture.fromJson(Map<String, dynamic> json) {
    return Culture(
      id: json['id'] as int,
      nomCommun: json['nom_commun'] as String? ?? '',
      nomScientifique: json['nom_scientifique'] as String?,
      type: json['type'] as String?,
      saison: json['saison'] as String?,
      region: json['region'] as String?,
      soilType: json['soil_type'] as String?,
      phSolMin: (json['ph_sol_min'] as num?)?.toDouble(),
      phSolMax: (json['ph_sol_max'] as num?)?.toDouble(),
      tempMin: (json['temp_min'] as num?)?.toDouble(),
      tempMax: (json['temp_max'] as num?)?.toDouble(),
      besoinEauCycle: (json['besoin_eau_cycle'] as num?)?.toDouble(),
      conseils: json['conseils'] as String?,
      products: (json['products'] as List?)
              ?.map((e) => Product.fromJson(e as Map<String, dynamic>))
              .toList() ??
          const [],
    );
  }

  @override
  List<Object?> get props => [id, nomCommun, type, saison];
}
